WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -46,25 +46,42 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+$.fn.goodDate = function() {
+  this.parent().siblings('p').removeClass('alert alert-danger');
+  this.removeClass('alert-danger');
+};
+
+$.fn.badDate = function() {
+  this.parent().siblings('p').addClass('alert alert-danger');
+  this.parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
+  this.addClass('alert-danger');
+};
+
 $(function() {
   $('.dropdown-toggle').dropdownHover();
   $('form.are-you-sure').areYouSure();
 
   // Copied (and modified by jfly) from
-  //  https://github.com/zpaulovics/datetimepicker-rails
+  // https://github.com/zpaulovics/datetimepicker-rails
   $('.date_picker.form-control, .datetime_picker.form-control').datetimepicker({
     useStrict: true, keepInvalid: true, useCurrent: false
   });
 
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.error', function(e){
-    $(this).parent().siblings('p').addClass('alert alert-danger');
-    $(this).parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
-    $(this).addClass('alert-danger');
-  });
-
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.change', function(e){
-    $(this).parent().siblings('p').removeClass('alert alert-danger');
-    $(this).removeClass('alert-danger');
+  $('.date_picker.form-control').on('blur', function(){
+    var date = $(this).val();
+    if(date.match(/^\d{4}-((0\d)|(1[012]))-(([012]\d)|3[01])$/)) {
```

Comment:
Silly me...it actually checks the values too. Didn't see that one coming.

But as I said, if you start with date X and just add an "a", it will not trigger dp.change, so we can't use that (we can, since it would cover the "real" usage, but...)